.gitlab-ci.yml

```diff
@@ -212,6 +212,25 @@ test-lin-ancient-cran:
     # Restore checking vignettes if upgrading our R dependency means knitr can be installed.
     - R CMD check --no-manual --no-build-vignettes --ignore-vignettes $(ls -1t data.table_*.tar.gz | head -n 1)
 
+# run the main checks with Address(+Leak),UBSanitizer enabled
+test-lin-san:
+  <<: *test-lin
+  image: docker.io/rocker/r-devel-ubsan-clang
```

Comment:
is this `r-devel-gcc-strict` image equivalent?

https://gitlab.com/jangorecki/dockerfiles/-/blame/master/r-devel/r-devel-gcc-strict?ref_type=heads#L43

(thinking it may be preferable to consistently use Jan's repo)

Reply:
This way the entire R and (almost) all packages are compiled with `-fsanitize=address,undefined`. I think there is a way compile just `data.table` with sanitizers by setting `PKG_CFLAGS`, `PKG_LIBS`, but then either the sanitizers must be manually preloaded using `LD_PRELOAD` or R must be linked with custom `MAIN_LD`(`FLAGS`) set.